### PR TITLE
#161878004 Create endpoint to delete a single comment

### PR DESCRIPTION
**Pull request by PriscillaSam**
#### What does this PR do?
Create endpoint to soft delete a comment. **DELETE** `/api/v1/articles/:slug/comments/:id`.

#### Description of Task to be completed
- write function to delete a comment
- write controller function for endpoint
- add swagger documentation for endpoint

#### How should this be manually tested?
- on the terminal run `git clone https://github.com/andela/haven-ah-backend.git`.
- run  `git checkout ft-delete-comment-161878004`.
- ensure that PostgreSQL is running on your local machine.
- set up databases (test and development) following the .env.sample file.
- run `npm install` to install the npm packages.
- run `npm test` to run the unit and integration tests.
To test on postman

- run `npm run start-dev` and open up postman. 

Authentication is required to perform this operation.
signup (`localhost:5000/api/v1/users/signup`) and use the email and password to log in (`localhost:5000/api/v1/users/signin`). Copy token and pass to the x-access-token header to delete a comment.

#### Screenshots where applicable
_Sample request url_
screen shot 2018-11-15 at 2 03 17 pm

_Response when comment does not exist_
screen shot 2018-11-15 at 1 21 10 pm

_Response when a user not an admin tries to delete a comment by another user_
screen shot 2018-11-15 at 1 33 36 pm

_Response when comment is accessed via the wrong article_
screen shot 2018-11-15 at 1 34 25 pm

_Response when delete operation is successful_
screen shot 2018-11-15 at 5 41 08 pm

#### What are the relevant Pivotal Tracker stories
#161878004

### CODE REVIEW

File: .vscode/launch.json [hunk 1]
```diff
@@ -0,0 +1,13 @@
+{
```
Comment (seyi-adeleke) [inline .vscode/launch.json]: please remove this

File: server/controllers/comment.js [hunk 2]
```diff
@@ -117,6 +117,52 @@ class Comment {
...
+
+    if (comment.articleId !== article.id) {
+      return badHttpResponse(
+        response,
+        400,
+        'Sorry, this comment is being accessed wrongly.'
+      );
+    }
+
+    if (role === 'user' && userId !== comment.userId) {
```
Comment (seyi-adeleke) [inline server/controllers/comment.js:178]: an admin should also be able to delete a comment

File: server/controllers/comment.js [hunk 3]
```diff
@@ -117,6 +117,52 @@ class Comment {
...
+        'Sorry, you can not perform this operation.'
+      );
+    }
+
+    const deleted = await commentRepo.deleteComment(comment);
+    return goodHttpResponse(
+      response,
+      200,
+      'Successfully deleted comment.',
+      { deletedComment: deleted }
```
Comment (seyi-adeleke) [inline server/controllers/comment.js]: no need to return the deleted object here

**Review by seyi-adeleke — CHANGES_REQUESTED** [review]

File: .vscode/launch.json (hunk 1 above)
Comment (ProfJigsaw) [inline .vscode/launch.json]: This shoudn't be on GitHub. Add this to your `.gitignore`

File: server/controllers/comment.js (hunk 2 above)
Comment (PriscillaSam) [reply server/controllers/comment.js:178]: @seyi-adeleke yes they are allowed. This check here is for a user with a user role trying to delete a comment by another user.

File: server/controllers/comment.js (hunk 3 above)
Comment (PriscillaSam) [reply server/controllers/comment.js]: @seyi-adeleke Noted.

**Review by seyi-adeleke — APPROVED** [review]

**Review by seyi-adeleke — APPROVED** [review]

**Comment by TheoOkafor** [comment]
Nice work @PriscillaSam

**Pull request merged**